### PR TITLE
Refactor build script and improve lint script

### PR DESCRIPTION
**Pull request by DanielRamosAcosta**
I've refactored the build script with some new sugar. Now the lint script reads from the original source instad of the built theme.

I've also moved the scripts into a `scripts` directory so top level files doesn't get too noisy.

### CODE REVIEW

**Comment by dsifford** [comment]
Sorry for the delay @DanielRamosAcosta!

Digging into this right now :+1:

File: scripts/build.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+'use strict';
```
[the remark this answers is not in the document]
Comment (DanielRamosAcosta) [reply scripts/build.js]: It enables strict mode the same way it does in the browser. This is avoiding undeclared variables to be global, leaking `this` into the global object...

File: scripts/yaml.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+const withAlphaType = new Type('!alpha', {
+    kind: 'sequence',
+    construct: ([hexRGB, alpha]) => hexRGB + alpha,
+    represent: ([hexRGB, alpha]) => hexRGB + alpha,
+});
+
+const schema = Schema.create([withAlphaType]);
+
+async function loadYAML(file) {
```
[the remark this answers is not in the document]
Comment (DanielRamosAcosta) [reply scripts/yaml.js]: I splited into two files just for separating concerns of the YAML related things and the actual theme load.

File: scripts/loadThemes.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    const standardThemeYAML = await fsp.readFile(yamlFilePath, 'utf8');
+    const standardTheme = await loadYAML(standardThemeYAML);
+
+    const softThemeYAML = getSoftThemeYAML(standardThemeYAML, standardTheme);
+    const sotfTheme = await loadYAML(softThemeYAML);
+
+    return { standardTheme, sotfTheme };
+}
+
+function getSoftThemeYAML(fileContent, standardTheme) {
```
[the remark this answers is not in the document]
Comment (DanielRamosAcosta) [reply scripts/loadThemes.js]: I separated this into a function for indicating that the soft theme needs being computed and the standard no (so the standard doesn't have it's own function).

But yes, maybe its very complex. I'll try using the original regex approach.

**Comment by DanielRamosAcosta** [comment]
This night I'll try to push some changes you've requested 😄

File: scripts/loadThemes.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (DanielRamosAcosta) [reply scripts/loadThemes.js]: Ok, I've refactored this function, take a look now 😉

File: scripts/loadThemes.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+const fsp = require('./fsp');
+const tinycolor = require('tinycolor2');
+const { loadYAML } = require('./yaml');
+
+async function loadTheme(yamlFilePath) {
+    const standardThemeYAML = await fsp.readFile(yamlFilePath, 'utf8');
+    const standardTheme = await loadYAML(standardThemeYAML);
+
+    const softThemeYAML = getSoftThemeYAML(standardThemeYAML, standardTheme);
+    const sotfTheme = await loadYAML(softThemeYAML);
```
[the remark this answers is not in the document]
Comment (DanielRamosAcosta) [reply scripts/loadThemes.js]: Fixed! 😄

**Comment by DanielRamosAcosta** [comment]
Any updates on this @dsifford?

**Comment by dsifford** [comment]
Ahh! So sorry about the delay. Completely forgot about this...

Let me look this over again to catch myself back up and if all checks out we should be good to go :+1:

File: scripts/build.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+const fsp = require('./fsp');
```
Comment (dsifford) [inline scripts/build.js]:
```suggestion
const path = require('path');
```

File: scripts/build.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+const path = require('path');
```
Comment (dsifford) [inline scripts/build.js]:
```suggestion
const fsp = require('./fsp');
```

File: scripts/lint.js [hunk 7]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
 const https = require('https');
-const fs = require('fs');
-const util = require('util');
+const loadThemes = require('./loadThemes');
```
Comment (dsifford) [inline scripts/lint.js]:
```suggestion
const path = require('path');
```

File: scripts/lint.js [hunk 8]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
 const https = require('https');
-const fs = require('fs');
-const util = require('util');
+const loadThemes = require('./loadThemes');
+const path = require('path');
```
Comment (dsifford) [inline scripts/lint.js]:
```suggestion
const loadThemes = require('./loadThemes');
```

File: scripts/loadThemes.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const fsp = require('./fsp');
```
Comment (dsifford) [inline scripts/loadThemes.js]:
```suggestion
const tinycolor = require('tinycolor2');
```

File: scripts/loadThemes.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const fsp = require('./fsp');
+const tinycolor = require('tinycolor2');
```
Comment (dsifford) [inline scripts/loadThemes.js]:
```suggestion
const fsp = require('./fsp');
```

File: scripts/build.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+    const { standardTheme, softTheme } = await loadThemes(THEME_YAML_FILE);
+    const standardThemePath = path.join(THEME_DIR, 'dracula.json');
+    const softThemePath = path.join(THEME_DIR, 'dracula-soft.json');
+
+    await Promise.all([
+        fsp.writeFile(standardThemePath, toJSON(standardTheme)),
+        fsp.writeFile(softThemePath, toJSON(softTheme)),
+    ]);
+})().catch(err => console.error(err))
```
Comment (dsifford) [inline scripts/build.js]:
```suggestion
})();
```

We want it to exit abruptly and fail here if the build fails so prob should just scrap the catch.

**Comment by DanielRamosAcosta** [comment]
Alright @dsifford! I think I fixed all the requested changes. I didn't use Github's *Apply Suggestion* feature because this would have generated like seven commits...

**Comment by dsifford** [comment]
Cool, looks good! Thanks a bunch!

(Re: 7 commits --- I squash and merge so no worries!)

**Pull request merged**